res/res_pjsip_sdp_rtp.c

```diff
@@ -1969,14 +2026,23 @@ static int create_outgoing_sdp_stream(struct ast_sip_session *session, struct as
 				continue;
 			}
 
-			if ((attr = generate_rtpmap_attr(session, media, pool, rtp_code, 0, NULL, index))) {
-				media->attr[media->attr_count++] = attr;
+			if (index != AST_RTP_DTMF) {
+				if ((attr = generate_rtpmap_attr(session, media, pool, rtp_code, 0, NULL, index))) {
+					media->attr[media->attr_count++] = attr;
+				}
 			}
-
-			if (index == AST_RTP_DTMF) {
-				snprintf(tmp, sizeof(tmp), "%d 0-16", rtp_code);
-				attr = pjmedia_sdp_attr_create(pool, "fmtp", pj_cstr(&stmp, tmp));
-				media->attr[media->attr_count++] = attr;
+			else {
+				/* Walk through the possible bitrates for the 2833/4733 digits and add them manually */
+				for (int i = 0; i < AST_RTP_NUM_DTMF_BITRATES; i++) {
+					if (bitRates[i] != 0) {
+						if ((attr = generate_rtpmap_attr2(session, media, pool, (rtp_code+i), 0, NULL, index, bitRates[i]))) {
```

Comment:
The RTP payload code assignment needs explanation here before I can really understand it. Code here should not be doing assignment itself of an RTP payload code, or assuming one using rtp_code+i. It should be retrieved from the RTP Engine core/codecs stuff.